### PR TITLE
feat(agent): add log level config option

### PR DESCRIPTION
**Pull request by mfulb**
NOTE: This is a reposting of an existing PR which "bduranleau-nr" created the vast majority of the work.    I did some work to tweak it and add tests and when I needed to rebase the branch it seemed better to make a new PR than to overwrite his original work/branch with a force push so we could have it as a reference

This PR addresses adding the ability for a user to configure a "minimum" log level to forward to the collector, based on the following PHP Log Level standards:
- https://datatracker.ietf.org/doc/html/rfc5424#section-6.2.1
- https://www.php-fig.org/psr/psr-3/#5-psrlogloglevel

### CODE REVIEW

**Comment by mfulb** [comment]
ok jenkins

**Comment by mfulb** [comment]
ok jenkins

**Review by misiekn — APPROVED** [review]
Looks good! 👍

**Comment by zsistla** [comment]
Looks like it has a conflict that needs to be resolved:
image

File: axiom/nr_log_level.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+#define LL_EMER_STR "EMERGENCY"
+#define LL_ALER_STR "ALERT"
+#define LL_CRIT_STR "CRITICAL"
+#define LL_ERRO_STR "ERROR"
+#define LL_WARN_STR "WARNING"
+#define LL_NOTI_STR "NOTICE"
+#define LL_INFO_STR "INFO"
+#define LL_DEBU_STR "DEBUG"
+#define LL_UNKN_STR "UNKNOWN"
```
Comment (zsistla) [inline axiom/nr_log_level.h:36]: A small (not enough to hold back merging) nit about the inconsistent naming convention prefix: `LL` vs `LOG_LEVEL`

**Review by zsistla — APPROVED** [review]
This is a great feature, thanks.

**Review by ZNeumann — APPROVED** [review]

**Pull request merged**